### PR TITLE
Add species report view

### PR DESCRIPTION
**Pull request by frwickst**
image

Just for fun I did a small experiment where I used the iNaturalist API for fetching an image of the animal that was shown 😄Note that the code below is not included in the PR. If you want to try it out, paste the following code into the HTML. If the City of Helsinki has access to some other image source maybe we could include that

```
<script>
    document.addEventListener("DOMContentLoaded", function () {
        fetch('http://api.inaturalist.org/v1/taxa?q={{ species.name_sci_1 }}')
            .then(function (response) {
                return response.json();
            })
            .then(function (response) {
                if (response.total_results >= 1) {
                    const animal = response.results[0];
                    if (!animal.default_photo) {
                        return false;
                    }

                    const container = document.getElementsByClassName('container')[0];
                    const image = document.createElement("img");
                    image.src = animal.default_photo.medium_url;
                    container.prepend(image)
                }
            });
    });
</script>
```
Here is what is looks like with the script included:
image

Refs https://github.com/City-of-Helsinki/ltj/issues/115

### CODE REVIEW

**Comment by frwickst** [comment]
@mingfeng DO NOT MERGE YET!

I just noticed that the translations are still missing.

File: nature/templates/nature/species-report.html [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                </div>
+            </div>
+            {% if not forloop.last %}
+                <hr class="mb-2 dashed">
+            {% endif %}
+        {% endfor %}
+        <hr class="mb-4">
+    {% endif %}
+
+    <!-- Regulations -->
```
Comment (mingfeng) [inline nature/templates/nature/species-report.html]: I guess these are features?

File: nature/views.py [hunk 2]
```diff
@@ -6,14 +6,34 @@
...
+            for observation in self.object.observations.all():
+                feature_class = observation.feature.feature_class
+                if feature_class.id not in context['feature_classes'].keys():
+                    context['feature_classes'][feature_class.id] = {
+                        'name': feature_class.name,
+                        'observations': [],
+                    }
+                context['feature_classes'][feature_class.id]['observations'].append(observation)
+        context.update(kwargs)
+        return super().get_context_data(**context)
```
Comment (mingfeng) [inline nature/views.py]: I think it's generally a good idea to call `get_context_data` base implementation first and then update the context in child class. This way the context variable overriding becomes explicit.

Comment (frwickst) [reply nature/views.py]: I'm following the structure that Django is using the for the function here. They are calling the super call last here as well. That is why I'm doing this.

Django implementation:
```
    def get_context_data(self, **kwargs):
        """
        Insert the single object into the context dict.
        """
        context = {}
        if self.object:
            context['object'] = self.object
            context_object_name = self.get_context_object_name(self.object)
            if context_object_name:
                context[context_object_name] = self.object
        context.update(kwargs)
        return super(SingleObjectMixin, self).get_context_data(**context)
```

Comment (frwickst) [reply nature/views.py]: But I can flip it around and do it the other way.

Comment (mingfeng) [reply nature/views.py]: The base class of SingleObjectMixin will check if the view context variable is already set and only set it when it's not available, so I think it's fine there. But other views bases generally do not do this.

**Pull request merged**